lib/QubitFlatfileImport.class.php

```diff
@@ -962,6 +992,104 @@ private function setInformationObjectByKeymap($legacyId)
   }
 
   /**
+   * Handle various update options when importing repositories and actors.
+   *
+   * @return bool  Whether to skip row processing for this record.
+   */
+  private function handleRepositoryAndActorRow()
+  {
+    // Not updating and not skipping matches: create a new record without checking
+    if (!$this->isUpdating() && !$this->skipMatched)
+    {
+      $this->object = new $this->className;
+
+      return false;
+    }
+
+    // Check existing repo by auth. form of name
+    $query = "SELECT object.id
+      FROM object JOIN actor_i18n i18n
+      ON object.id = i18n.id
+      WHERE i18n.authorized_form_of_name = ?
+      AND object.class_name = ?;";
+
+    $statement = QubitFlatfileImport::sqlQuery($query, array($this->columnValue('authorizedFormOfName'), $this->className));
```

Comment:
Hi @MikeFE, I like the idea, but right now it's being used 11 times in QubitFlatfileImport.class.php, I'd rather change it when we replace it everywhere. Please, let me know if you're okay with it. 😉

And thanks for the recommendation, I'd not use it in the XML import changes I'm doing.